core_matcher.go

```diff
@@ -31,15 +31,12 @@ type coreMatcher struct {
 
 // coreFields groups the updateable fields in coreMatcher.
 // state is the start of the automaton.
-// namesUsed is a map of field names that are used in any of the patterns that this automaton encodes. Typically,
+// segmentsTree is a tree of segments that are used in any of the patterns that this automaton encodes.Typically,
 // patterns only consider a subset of the fields in an incoming data object, and there is no reason to consider
 // fields that do not appear in patterns when using the automaton for matching.
-// fakeField is used when the flattener for an event returns no fields, because it could still match if
```

Comment:
@timbray FYI - I removed the "fakeField" documentation since you removed it from the struct, hope it's ok.

Reply:
Thanks.